packages/botonic-plugin-flow-builder/src/action/intent.ts

```diff
@@ -15,14 +15,17 @@ export async function getIntentNodeByInput(
     confidence: request.input.confidence as number,
     confidence_successful: true,
   }
-  if (isIntentValid(intentNode, request, cmsApi) && intentNode?.target?.id) {
-    await trackEvent(request, EventName.botAiModel, eventArgs)
-    return intentNode
-  } else {
+  if (request.input.confidence && request.input.intent) {
+    if (isIntentValid(intentNode, request, cmsApi) && intentNode?.target?.id) {
```

Comment:
You can add this `&& intentNode?.target?.id` inside `isIntentValid` method

Reply:
This intentNode.target.id was here because of the logic I changed in the previous PR, it is not needed now. What would be logical is that if there is no IntentNode for the received request.input.intent then don't create event for botAiModel.

```
if (request.input.confidence && request.input.intent && intentNode) {
    if (isIntentValid(intentNode, request, cmsApi)) {
```